sourcemap file contains relative path calculated from parent of outDir and basePath

Hi from ng-conf! :)

TS nightly.

```
alexeagle-macbookpro2:repro_sourceMap alexeagle$ cat some/path/to/src/myfile.ts
export let a:string;
alexeagle-macbookpro2:repro_sourceMap alexeagle$ cat some/path/to/src/tsconfig.json
{
    "compilerOptions": {
        "module": "commonjs",
        "target": "es5",
        "noImplicitAny": false,
        "sourceMap": true,
        "inlineSources": true,
        "outDir": "../../../../dist"

    },
    "exclude": [
        "node_modules"
    ]
}
$ cat dist/myfile.js.map
{"version":3,"file":"myfile.js","sourceRoot":"","sources":["../some/path/to/src/myfile.ts"],"names":[],"mappings":";AAAoB","sourcesContent":["export let a:string;"]}
```

the sourceMap contains my private bits "some/path/to/" which we would like it not to.
This causes devTools in the browser to show a strange path that confuses users.

<img width="265" alt="screen shot 2016-05-03 at 12 52 03 pm" src="https://cloud.githubusercontent.com/assets/47395/14994603/f848909e-112d-11e6-9db0-85279115580f.png">

The obvious way to fix this would be the `sourceRoot` option in `tsconfig.json` which we could set to `..` but this is not allowed:

`error TS5053: Option 'sourceRoot' cannot be specified with option 'inlineSources'.`

Proposal is that sourceRoot should be allowed with inlineSources. The resulting `.js.map` file would contain a path that doesn't resolve locally on disk, but does work at runtime. (it's similar to the pathMapping feature where design-time filesystem layout does not need to match runtime)

cc @igorminar

--- a/src/compiler/diagnosticMessages.json
+++ b/src/compiler/diagnosticMessages.json
@@ -2260,7 +2260,7 @@
         "category": "Error",
         "code": 5047
     },
-    "Option 'inlineSources' can only be used when either option '--inlineSourceMap' or option '--sourceMap' is provided.": {
+    "Option '{0} can only be used when either option '--inlineSourceMap' or option '--sourceMap' is provided.": {
         "category": "Error",
         "code": 5051
     },

--- a/src/compiler/program.ts
+++ b/src/compiler/program.ts
@@ -2043,12 +2043,12 @@ namespace ts {
                 }
             }
 
-            if (options.inlineSources) {
-                if (!options.sourceMap && !options.inlineSourceMap) {
-                    programDiagnostics.add(createCompilerDiagnostic(Diagnostics.Option_inlineSources_can_only_be_used_when_either_option_inlineSourceMap_or_option_sourceMap_is_provided));
+            if (!options.sourceMap && !options.inlineSourceMap) {
+                if (options.inlineSources) {
+                    programDiagnostics.add(createCompilerDiagnostic(Diagnostics.Option_0_can_only_be_used_when_either_option_inlineSourceMap_or_option_sourceMap_is_provided, "inlineSources"));
                 }
                 if (options.sourceRoot) {
-                    programDiagnostics.add(createCompilerDiagnostic(Diagnostics.Option_0_cannot_be_specified_with_option_1, "sourceRoot", "inlineSources"));
+                    programDiagnostics.add(createCompilerDiagnostic(Diagnostics.Option_0_can_only_be_used_when_either_option_inlineSourceMap_or_option_sourceMap_is_provided, "sourceRoot"));
                 }
             }
 
@@ -2056,14 +2056,9 @@ namespace ts {
                 programDiagnostics.add(createCompilerDiagnostic(Diagnostics.Option_0_cannot_be_specified_with_option_1, "out", "outFile"));
             }
 
-            if (!options.sourceMap && (options.mapRoot || options.sourceRoot)) {
-                // Error to specify --mapRoot or --sourceRoot without mapSourceFiles
-                if (options.mapRoot) {
-                    programDiagnostics.add(createCompilerDiagnostic(Diagnostics.Option_0_cannot_be_specified_without_specifying_option_1, "mapRoot", "sourceMap"));
-                }
-                if (options.sourceRoot && !options.inlineSourceMap) {
-                    programDiagnostics.add(createCompilerDiagnostic(Diagnostics.Option_0_cannot_be_specified_without_specifying_option_1, "sourceRoot", "sourceMap"));
-                }
+            if (options.mapRoot && !options.sourceMap) {
+                // Error to specify --mapRoot without --sourcemap
+                programDiagnostics.add(createCompilerDiagnostic(Diagnostics.Option_0_cannot_be_specified_without_specifying_option_1, "mapRoot", "sourceMap"));
             }
 
             if (options.declarationDir) {

--- a/tests/baselines/reference/optionsSourcemapInlineSourcesSourceRoot.errors.txt
+++ b/tests/baselines/reference/optionsSourcemapInlineSourcesSourceRoot.errors.txt
@@ -1,7 +1,0 @@
-error TS5053: Option 'sourceRoot' cannot be specified with option 'inlineSources'.
-
-
-!!! error TS5053: Option 'sourceRoot' cannot be specified with option 'inlineSources'.
-==== tests/cases/compiler/optionsSourcemapInlineSourcesSourceRoot.ts (0 errors) ====
-    
-    var a = 10;

--- a/tests/baselines/reference/optionsSourcemapInlineSourcesSourceRoot.symbols
+++ b/tests/baselines/reference/optionsSourcemapInlineSourcesSourceRoot.symbols
@@ -1,0 +1,5 @@
+=== tests/cases/compiler/optionsSourcemapInlineSourcesSourceRoot.ts ===
+
+var a = 10;
+>a : Symbol(a, Decl(optionsSourcemapInlineSourcesSourceRoot.ts, 1, 3))
+

--- a/tests/baselines/reference/optionsSourcemapInlineSourcesSourceRoot.types
+++ b/tests/baselines/reference/optionsSourcemapInlineSourcesSourceRoot.types
@@ -1,0 +1,6 @@
+=== tests/cases/compiler/optionsSourcemapInlineSourcesSourceRoot.ts ===
+
+var a = 10;
+>a : number
+>10 : number
+

--- a/tests/baselines/reference/project/mapRootSourceRootWithNoSourceMapOption/amd/mapRootSourceRootWithNoSourceMapOption.errors.txt
+++ b/tests/baselines/reference/project/mapRootSourceRootWithNoSourceMapOption/amd/mapRootSourceRootWithNoSourceMapOption.errors.txt
@@ -1,9 +1,9 @@
+error TS5051: Option 'sourceRoot can only be used when either option '--inlineSourceMap' or option '--sourceMap' is provided.
 error TS5052: Option 'mapRoot' cannot be specified without specifying option 'sourceMap'.
-error TS5052: Option 'sourceRoot' cannot be specified without specifying option 'sourceMap'.
 
 
+!!! error TS5051: Option 'sourceRoot can only be used when either option '--inlineSourceMap' or option '--sourceMap' is provided.
 !!! error TS5052: Option 'mapRoot' cannot be specified without specifying option 'sourceMap'.
-!!! error TS5052: Option 'sourceRoot' cannot be specified without specifying option 'sourceMap'.
 ==== m1.ts (0 errors) ====
     var m1_a1 = 10;
     class m1_c1 {

--- a/tests/baselines/reference/project/mapRootSourceRootWithNoSourceMapOption/node/mapRootSourceRootWithNoSourceMapOption.errors.txt
+++ b/tests/baselines/reference/project/mapRootSourceRootWithNoSourceMapOption/node/mapRootSourceRootWithNoSourceMapOption.errors.txt
@@ -1,9 +1,9 @@
+error TS5051: Option 'sourceRoot can only be used when either option '--inlineSourceMap' or option '--sourceMap' is provided.
 error TS5052: Option 'mapRoot' cannot be specified without specifying option 'sourceMap'.
-error TS5052: Option 'sourceRoot' cannot be specified without specifying option 'sourceMap'.
 
 
+!!! error TS5051: Option 'sourceRoot can only be used when either option '--inlineSourceMap' or option '--sourceMap' is provided.
 !!! error TS5052: Option 'mapRoot' cannot be specified without specifying option 'sourceMap'.
-!!! error TS5052: Option 'sourceRoot' cannot be specified without specifying option 'sourceMap'.
 ==== m1.ts (0 errors) ====
     var m1_a1 = 10;
     class m1_c1 {

--- a/tests/baselines/reference/project/sourceRootWithNoSourceMapOption/amd/sourceRootWithNoSourceMapOption.errors.txt
+++ b/tests/baselines/reference/project/sourceRootWithNoSourceMapOption/amd/sourceRootWithNoSourceMapOption.errors.txt
@@ -1,7 +1,7 @@
-error TS5052: Option 'sourceRoot' cannot be specified without specifying option 'sourceMap'.
+error TS5051: Option 'sourceRoot can only be used when either option '--inlineSourceMap' or option '--sourceMap' is provided.
 
 
-!!! error TS5052: Option 'sourceRoot' cannot be specified without specifying option 'sourceMap'.
+!!! error TS5051: Option 'sourceRoot can only be used when either option '--inlineSourceMap' or option '--sourceMap' is provided.
 ==== m1.ts (0 errors) ====
     var m1_a1 = 10;
     class m1_c1 {

--- a/tests/baselines/reference/project/sourceRootWithNoSourceMapOption/node/sourceRootWithNoSourceMapOption.errors.txt
+++ b/tests/baselines/reference/project/sourceRootWithNoSourceMapOption/node/sourceRootWithNoSourceMapOption.errors.txt
@@ -1,7 +1,7 @@
-error TS5052: Option 'sourceRoot' cannot be specified without specifying option 'sourceMap'.
+error TS5051: Option 'sourceRoot can only be used when either option '--inlineSourceMap' or option '--sourceMap' is provided.
 
 
-!!! error TS5052: Option 'sourceRoot' cannot be specified without specifying option 'sourceMap'.
+!!! error TS5051: Option 'sourceRoot can only be used when either option '--inlineSourceMap' or option '--sourceMap' is provided.
 ==== m1.ts (0 errors) ====
     var m1_a1 = 10;
     class m1_c1 {

